sdks/uniswapx-sdk/src/trade/PriorityOrderTrade.ts

```diff
@@ -0,0 +1,154 @@
+> {
+  public readonly tradeType: TTradeType;
+  public readonly order: UnsignedPriorityOrder;
+
+  private _inputAmount: CurrencyAmount<TInput> | undefined;
+  private _outputAmounts: CurrencyAmount<TOutput>[] | undefined;
+
+  private _currencyIn: TInput;
+  private _currenciesOut: TOutput[];
+
+  public constructor({
+    currencyIn,
+    currenciesOut,
+    orderInfo,
+    tradeType,
+  }: {
+    currencyIn: TInput;
+    currenciesOut: TOutput[];
+    orderInfo: UnsignedPriorityOrderInfo;
+    tradeType: TTradeType;
+  }) {
+    this._currencyIn = currencyIn;
+    this._currenciesOut = currenciesOut;
+    this.tradeType = tradeType;
+
+    // assume single-chain for now
+    this.order = new UnsignedPriorityOrder(orderInfo, currencyIn.chainId);
+  }
+
+  public get inputAmount(): CurrencyAmount<TInput> {
+    if (this._inputAmount) return this._inputAmount;
+
+    const amount = CurrencyAmount.fromRawAmount(
+      this._currencyIn,
+      this.order.info.input.amount.toString()
+    );
+    this._inputAmount = amount;
+    return amount;
+  }
+
+  public get outputAmounts(): CurrencyAmount<TOutput>[] {
+    if (this._outputAmounts) return this._outputAmounts;
+
+    const amounts = this.order.info.outputs.map((output) => {
+      // assume single chain ids across all outputs for now
+      const currencyOut = this._currenciesOut.find((currency) =>
+        areCurrenciesEqual(currency, output.token, currency.chainId)
+      );
+
+      if (!currencyOut) {
+        throw new Error("currency not found in output array");
+      }
+
+      return CurrencyAmount.fromRawAmount(
+        currencyOut,
+        output.amount.toString()
+      );
+    });
+
+    this._outputAmounts = amounts;
+    return amounts;
+  }
+
+  private _firstNonFeeOutputAmount:
+    | CurrencyAmount<TOutput>
+    | undefined;
+
+  private getFirstNonFeeOutputAmount(): CurrencyAmount<TOutput> {
+    if (this._firstNonFeeOutputAmount)
+      return this._firstNonFeeOutputAmount;
+
+    if (this.order.info.outputs.length === 0) {
+      throw new Error("there must be at least one output token");
+    }
+    const output = this.order.info.outputs[0];
+
+    // assume single chain ids across all outputs for now
+    const currencyOut = this._currenciesOut.find((currency) =>
+      areCurrenciesEqual(currency, output.token, currency.chainId)
+    );
+
+    if (!currencyOut) {
+      throw new Error(
+        "currency output from order must exist in currenciesOut list"
+      );
+    }
+
+    const amount =
+      CurrencyAmount.fromRawAmount(
+        currencyOut,
+        output.amount.toString()
+      );
+
+    this._firstNonFeeOutputAmount = amount;
+    return amount;
+  }
+
+  // TODO: revise when there are actually multiple output amounts. for now, assume only one non-fee output at a time
+  public get outputAmount(): CurrencyAmount<TOutput> {
+    // TODO: estimate epected amount, using classic quote or expected priority
+    return this.getFirstNonFeeOutputAmount();
```

Comment:
given amount in quote.info.outputs (assuming exact_in trade) is classic adjusted for gas , we can theoretically calculate the classic price here. Agree it's better to be able to pass in classic price, which requires a URA change (to add that info to the quote response) and was what I was planning to do

Reply:
Ah true, that won't be precise tho right since gas could be different from quote to quote?